src/cmd/cloud.js

```diff
@@ -690,8 +689,7 @@ module.exports = class CloudCommand extends CLICommandBase {
 		for (const file of files) {
 			const propPath = path.join(file, 'project.properties');
 			try {
-				const savedProp = await fs.readFile(propPath, 'utf8');
-				const savedPropObj = propertiesParser.parse(savedProp);
+				const savedPropObj = await utilities.parsePropertyFile(propPath);
```

Comment:
Given that now we have this new property to get assets directory, is this function needs to check if the directory path?
I saw in `_getAssets` function we are doing the same thing and handling in a better way the errors:
https://github.com/particle-iot/particle-cli/pull/650/files#diff-2a5669d3f56c5e923bddc240ef52da8109cb697d4c8707528f098b505b466796R80

I'd say just make either this function returns the path if we have the project property `assetOtaFolder` and let _getAssets do the validations (that now are happening) or remove the validations from _getAssets and put those in here.

Also I saw that in some places we use files[0] and corresponds with the tests that you put. Which is the use case where files will be more than one in this specific function? 

Here's the thing I'm talking about with files[0] 👉  https://github.com/particle-iot/particle-cli/pull/650/files#diff-94014247d62bc2951e9e7135b57b09171b6570579119026b858b7772feb84d98L336

Reply:
@hugomontero I get your point. I changed the code so that the error checking is done in _getAssets and we don't worry about it in the cloud's _checkForAssets. It's basically only checking if it's a directory. Nice optimization suggestion. As far as the files[0], I didn't mess with this and has been there. I am not exactly sure why we expect multiple arguments (i think for multiple source files?) but looks like the expectation is that the first argument in that list is the project directory to the specified by the user.